### ISSUE TITLE
Wrong Bit resetting

### ISSUE THREAD

**Issue by Thomas131** [zschroeder6212/tiny-i2c-spi#3]
Hi!
I am currently looking at the source code as inspiration for another project and I am very confused:

Looking at 
https://github.com/zschroeder6212/tiny-i2c-spi/blob/a4e1b82dc9b015b7c0f7bea63cb24e0dbc9753a4/src/SoftSPI.c#L55

That Line should do nothing in my understanding - you can bit-shift a 0 however you like, you will end up with 0b00000000.
And `0x?? | 0x00` will leave the value completly untouched afaik.

AFAIK setting a bit to zero is:
```C
REGISTERx &= ~(1<<Bit);
```

Best regards!
Thomas

**Comment by egormanga** [comment]
There are also some major issues with the main code (bit-shifting THEN masking the SPI parameters). I have emailed the repo owner earlier, got no answer.

**Comment by zschroeder6212** [comment]
Agreed, this is odd. I didn't look into the SPI code much since it seemed to work fine, but it could probably use a refactor

**Comment by egormanga** [comment]
> it seemed to work fine

With the sensor, it does not. Could you please look up and answer to my email from sdoregor@sdore.me?

**Comment by zschroeder6212** [comment]
I don't see any email from you, but I can assure you that it does work with the sensor. I have tested extensively.

If you are having problems then I would be happy to help troubleshoot, but it should probably be a separate issue.

**Comment by egormanga** [comment]
@zschroeder6212 Okay, I've just resent it to you at zschroeder6212@gmail.com (if that is the right address).
